### DISCUSSION TITLE
Stand-alone mode

### DISCUSSION THREAD

**Discussion by iincer** [pacti-org/pacti#135] [General]
Pacti can be used in two ways: as a python package, which is what we are all using, and as a standalone executable. When you install Pacti, you get access to the executable `pacti` and can run commands of the form
```
pacti input.json output.json
```
This input file, which should probably be called `input.pacti`, contains the contracts upon which Pacti operates, and the output contains the results.
How important do we think is to have a standalone executable? Having a standalone executable seems useful for someone who does not use python and wants to use Pacti.

**Comment by ayush9pandey** [comment]
Just stumbled upon the discussions on Github!

I like this idea. Agree that there should be a file extension called `filename.pacti` that has the string constructor for contracts (instead of JSON) and the output is printed out in the same format to a `.pacti` file.

**Reply by iincer** [reply]
Glad we are on the same page, @ayush9pandey   :-)

@NicolasRouquette, what are your thoughts about this, i.e., having a Pacti executable? (We have one now.) Since you/your colleagues often program in other languages, an executable may be the easiest way of using Pacti in a language-indepedent fashion.

**Comment by iincer** [comment]
Update: we are dropping standalone mode--see #156
We are still developing support for file IO.
